src/Assignments/ProblemIframe.tsx

```diff
@@ -60,42 +89,46 @@ export const ProblemIframe: React.FC<ProblemIframeProps> = ({
         setRenderedHTML('');
         // srcdoc='' triggers onLoad with setLoading(false) so setLoading(true) isn't effective until now
         setLoading(true); 
-        (async () => {
-            try {
-                let queryString = qs.stringify(_({
-                    workbookId,
-                    readonly
-                }).omitBy(_.isUndefined).value());
-                if (!_.isEmpty(queryString)) {
-                    queryString = `?${queryString}`;
-                }
 
-                let res;
-                if (previewPath || previewProblemSource) {
-                    res = await postPreviewQuestion({
-                        webworkQuestionPath: previewPath,
-                        problemSeed: previewSeed,
-                        problemSource: previewProblemSource,
-                        showHints: previewShowHints,
-                        showSolutions: previewShowSolutions
-                    });
-                } else {
-                    res = await AxiosRequest.get(`/courses/question/${problem.id}${queryString}`);
-                }
-                // TODO: Error handling.
-                setRenderedHTML(res.data.data.rendererData.renderedHTML);
-            } catch (e) {
-                setError(e.message);
-                logger.error('Error posting preview', e, e.message);
-                setLoading(false);
-            }
-        })();
+        fetchHTML();
+
         // when problem changes, reset lastsubmitted and lastsaved
         setLastSubmittedAt?.(null);
         setLastSavedAt?.(null);
         setLastSubmission({});
     }, [problem, problem.id, workbookId, previewPath, previewProblemSource, previewSeed]);
 
+    const getHTML = async () => {
+        try {
+            let queryString = qs.stringify(_({
+                workbookId,
+                readonly
+            }).omitBy(_.isUndefined).value());
+            if (!_.isEmpty(queryString)) {
```

Comment:
Nit: I don't know if we areally need an empty check here, having an empty query string should be fine